writer.go

```diff
@@ -212,13 +212,13 @@ func (w *Writer) DoRemove() {
 	}
 }
 
-// CompressFile compress log file write into .gz and remove source file
+// CompressFile compress log file write into .gz
 func (w *Writer) CompressFile(oldfile *os.File, cmpname string) error {
 	cmpfile, err := os.OpenFile(cmpname, DefaultFileFlag, DefaultFileMode)
-	defer cmpfile.Close()
```

Comment:
file should be closed after use

Reply:
see the `defer` added 2 lines below, if there is an error we shouldn't call defer afaik